incubator/monochart/templates/statefulset.yaml

```diff
@@ -0,0 +1,85 @@
+{{- if .Values.statefulset.enabled -}}
+kind: StatefulSet
+apiVersion: apps/v1beta1
+metadata:
+  name: {{ include "common.fullname" . }}
+{{- with .Values.statefulset.annotations }}
+  annotations:
+{{ toYaml . | indent 4 }}
+{{- end }}
+  labels:
+{{ include "common.labels.standard" . | indent 4 }}
+spec:
+  selector:
+    matchLabels:
+      app: {{ include "common.name" . }}
+      release: {{ .Release.Name }}
+  serviceName: {{ include "common.fullname" . }}
+  replicas: 1
+{{- with .Values.statefulset.strategy }}
+  updateStrategy:
+{{ toYaml . | indent 4 }}
+{{- end }}
+  revisionHistoryLimit: {{ .Values.statefulset.revisionHistoryLimit }}
+  template:
+    metadata:
+      name: {{ include "common.fullname" . }}
+{{- with .Values.statefulset.pod.annotations }}
+      annotations:
+{{ toYaml . | indent 8 }}
+{{- end }}
+      labels:
+        app: {{ include "common.name" . }}
+        release: {{ .Release.Name | quote }}
+    spec:
+      terminationGracePeriodSeconds: 0
+      containers:
+      - name: {{ .Chart.Name }}
+        image: {{ required "image.repository is required!" .Values.image.repository }}:{{ required "image.tag is required!" .Values.image.tag }}
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+{{ include "monochart.env" . | indent 8 }}
+        args: {}
+        ports:
+          - name: http
+            containerPort: 8080
```

Comment:
Parameterize?

Reply:
In newer version of helm (2.9.1 and higher) when your create chart via `helm create chart_name` it has some changes in service template and deployments ports and i thins it really reasonable changes. Service have only one parameterized port - external one, and containers port is hardcoded and with assigned name.
I'll explain.
Container port depends on docker image that you use therefore it is unlikely that it will ever be changed - so it can be hardcoded directly in deployment, but it shoud have a name `http` in this case. External port of the service is some kind of interface so it shoud be configurable.

Reply:
I think that makes sense for a custom built chart for a single purpose. But this is a universal chart so I think it’s better we parameteize it for more interoperability. Eg node apps default on 3000.